docs/performance/inserts/bulk.rst

```diff
@@ -0,0 +1,419 @@
+this how-to guide if you want to improve performance.
+
+Briefly, when importing data in bulk, keep the following in mind:
+
+- Reduce the number of replicas as much as possible (ideally zero) as
+  replication slows down the import process significantly
+
+- Only use as many shards as you need
+
+- Disable the periodic table refresh by setting the refresh interval to zero
+  during import
+
+- Adjust the bulk size of the import as necessary given the number of records
+  you are importing
+
+- Import table partitions separately using the ``PARTITION`` clause in the
+  ``COPY TO`` statement
+
+The rest of this document goes into more detail.
+
+.. TIP::
+
+    Increasing disk `IOPS`_ is the best, and perhaps most straightforward, way
+    to increase import speed. SSDs are recommended (as general good practice)
+    for this reason.
+
+    Another way to increase import speed is to add more disks by adding more
+    nodes. CrateDB is a `distributed database`_, and so, increasing overall
+    cluster size is generally a good way to improve performance.
+
+.. rubric:: Table of contents
+
+.. contents::
+    :local:
+
+.. SEEALSO::
+
+    :ref:`Insert performance: Methods <insert-methods>`
+
+    :ref:`Best practices: Migrating from MySQL <migrating-mysql>`
+
+    :ref:`Best practices: Migrating from MongoDB <migrating-mongodb>`
+
+    `CrateDB Reference: Import and export`_
+
+
+.. _bulk-configure-tables:
+
+Configure your tables
+=====================
+
+
+.. _bulk-data-structure:
+
+Data structure
+--------------
+
+Before starting the import, you need to carefully define your table structure
+in CrateDB. Decisions made at this point will influence the import process
+later.
+
+For this example, say you have a single ``users`` table with six columns of
+various types. You can create this table with the following statement::
+
+    cr> CREATE TABLE users (
+    ...   id INT primary key,
+    ...   name TEXT,
+    ...   day_joined TIMESTAMP,
+    ...   bio TEXT INDEX using fulltext,
+    ...   address OBJECT (dynamic) AS (
+    ...     city TEXT,
+    ...     country TEXT
+    ...   )
+    ... );
+    CREATE OK, 1 row affected (... sec)
+
+
+.. _bulk-shards-replicas:
+
+Shards and replicas
+-------------------
+
+For each table, if you do not set the number of `shards`_ or the number of
+`replicas`_, the default configuration is as follows:
+
+**Shards**
+    Dependant on the number of data-nodes in the cluster (see `CLUSTERED
+    clause`_)
+
+**Replicas**
+    One
+
+You should choose the number of shards wisely. The choice depends on the number
+of nodes in the cluster as well as on the amount of data that is going into the
+table.
+
+Assuming there are six nodes in the cluster, you can put two shards on each
+node, giving you a total of 12 shards. This should be enough to handle millions
+of records.
+
+Shards can be thought of as "virtual nodes." You should create enough for your
```

Comment:
```suggestion
Shards can be thought of as "virtual nodes". You should create enough for your
```

Reply:
this is incorrect. punctuation should be inside the quotes

https://www.grammarly.com/blog/quotation-marks/#:~:text=Do%20commas%20and%20periods%20go,go%20inside%2C%20sometimes%20stay%20outside.

Reply:
Thanks for improving my English writing along the lines and sorry for the noise.